.github/workflows/release.yml

```diff
@@ -0,0 +1,31 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    # should we check? what about merge commits?
+    # if: "contains(github.event.head_commit.message, 'version')"
+    name: Trigger a release
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1 # which version to use?
+        with:
+          ref: master
```

Comment:
why do I need to checkout to master, won't default be `master`? should I use `@v2`?

do I have to setup node? what are the situations where node needs to be setup, when you need to have a `run` step that needs node environment? is `npx` excluded bc it already has node in it?

Reply:
v2 can be used.

For this though, I’d suggest against checking out more than the latest HEAD commit of the master/main branch for simplicity. This way you only trigger a release action when master gets a commit that matches the message criteria of `versions - ...` (or whatever gets finalized or even configured through parameters), and not to worry about anything else such as retrospectively look back and compare if at most 3 versions commit were pushed — in which case, just create 3 releases, and let the tag-triggered action take care of the actual “release/deployment”